core/src/main/java/io/bisq/core/network/CoreSeedNodesRepository.java

```diff
@@ -1,58 +1,37 @@
+import javax.annotation.concurrent.NotThreadSafe;
 import javax.inject.Inject;
-import java.util.Arrays;
-import java.util.List;
 import java.util.Set;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-@Slf4j
+import static io.bisq.core.network.DefaultNodeAddresses.DEFAULT_LOCALHOST_SEED_NODE_ADDRESSES;
+import static io.bisq.core.network.DefaultNodeAddresses.DEFAULT_TOR_SEED_NODE_ADDRESSES;
+
+@NotThreadSafe
 public class CoreSeedNodesRepository implements SeedNodesRepository {
+    private final NodeAddressLookup lookup;
 
-    @Getter
-    private final Set<NodeAddress> seedNodeAddresses;
+    private Set<NodeAddress> seedNodeAddresses;
+    private Set<NodeAddress> torSeedNodeAddresses;
+    private Set<NodeAddress> localhostSeedNodeAddresses;
 
     @Inject
-    public CoreSeedNodesRepository(BisqEnvironment bisqEnvironment,
-                                   @Named(NetworkOptionKeys.USE_LOCALHOST_FOR_P2P) boolean useLocalhostForP2P,
-                                   @Named(NetworkOptionKeys.NETWORK_ID) int networkId,
-                                   @Nullable @Named(NetworkOptionKeys.MY_ADDRESS) String myAddress,
-                                   @Nullable @Named(NetworkOptionKeys.SEED_NODES_KEY) String seedNodes) {
-        List<String> bannedNodes = bisqEnvironment.getBannedSeedNodes();
-        Set<NodeAddress> nodeAddresses;
-        if (seedNodes != null && !seedNodes.isEmpty()) {
-            nodeAddresses = Arrays.asList(StringUtils.deleteWhitespace(seedNodes).split(","))
-                    .stream()
-                    .map(NodeAddress::new)
-                    .collect(Collectors.toSet());
-        } else {
-            nodeAddresses = useLocalhostForP2P ? localhostSeedNodeAddresses : torSeedNodeAddresses;
-            nodeAddresses = nodeAddresses.stream()
-                    .filter(e -> String.valueOf(e.getPort()).endsWith("0" + String.valueOf(networkId)))
-                    .collect(Collectors.toSet());
-        }
-
-        seedNodeAddresses = nodeAddresses.stream()
-                .filter(e -> myAddress == null || myAddress.isEmpty() || !e.getFullAddress().equals(myAddress))
-                .filter(e -> bannedNodes == null || !bannedNodes.contains(e.getHostName()))
-                .collect(Collectors.toSet());
+    public CoreSeedNodesRepository(NodeAddressLookup lookup) {
+        this.lookup = lookup;
+        this.torSeedNodeAddresses = DEFAULT_TOR_SEED_NODE_ADDRESSES;
+        this.localhostSeedNodeAddresses = DEFAULT_LOCALHOST_SEED_NODE_ADDRESSES;
+    }
 
-        if (bannedNodes == null)
-            log.info("seedNodeAddresses={}", seedNodeAddresses);
-        else
-            log.warn("We received banned seed nodes={}, seedNodeAddresses={}", bannedNodes, seedNodeAddresses);
+    @Override
+    public Set<NodeAddress> getSeedNodeAddresses() {
+        if (seedNodeAddresses == null) {
+            seedNodeAddresses = lookup.resolveNodeAddresses();
+        }
```

Comment:
Why the lazy initialization here in the setter? Is this an attempt to avoid an expensive operation? Is this actually an optional code path? If it is usually invoked, or if the operation is not terribly expensive, prefer assigning `seedNodeAddresses` directly in the constructor, such that the field can be marked `final`.

Reply:
`NodeAddressLookup` contains some non-trivial logic, so i thought that invoking it in constructor would not be a good idea. 

> prefer assigning seedNodeAddresses directly in the constructor, such that the field can be marked `final`

totally agree, that way is better. But it would lead us to creation of factory of `SeedNodesRepository`. Is it ok?